src/main/java/taxi/service/AuthenticationServiceIpl.java

```diff
@@ -0,0 +1,23 @@
+package taxi.service;
+
+import java.util.Optional;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Injector;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceIpl implements AuthenticationService {
+    private static final Injector injector = Injector.getInstance("taxi");
+    private final DriverService driverService = (DriverService) injector
```

Comment:
> Let’s inject driverService into AuthenticationServiceImpl class using @Inject annotation.

Reply:
But @Inject is using for fields and @Service is using for services

Reply:
You may inject DriverService just like any Dao object.

Reply:
```
@Service  public class AuthenticationServiceImpl implements AuthenticationService { 
      
     @Inject 
     private DriverService driverService;
```

Reply:
Oooohh I've got it